### ISSUE TITLE
HW acceleration not working in firefox 124.0.2

### ISSUE THREAD

**Issue by Vitaly76ASDgnaerkldbnak** [elFarto/nvidia-vaapi-driver#283]
HW acceleration not working in firefox 124.0.2, since FF self upgrade, I didn't notice from what version it broke, but in version 110 it still works normal. I check this on nvidia drivers V525.147.05 and V545.29.06. Option widget.dmabuf.force-enabled is true, but in 124.0.2 it is not helping. BTW, HW acceleration in MPV works fine.

**Comment by mirh** [comment]
Is there no difference in `about:support`?

**Comment by Vitaly76ASDgnaerkldbnak** [comment]
There are too many info in about:support, what exactly should I pay attention to?

**Comment by mirh** [comment]
The ones specifically about acceleration?
More or less everything in the graphics section, but decision log has the most important stuff.

**Comment by Vitaly76ASDgnaerkldbnak** [comment]
In section media, in ff 110 there are H264 HW and VP8 HW, but in FF 125 there are only SWs. It seems obvious, but why FF 125 is not detecting HW acceleration?

**Comment by mirh** [comment]
Ok, and that's the symptoms. But it's the section right before to be telling you the causes.

**Comment by LocalRaccoonEnthusiast** [comment]
![image](https://github.com/elFarto/nvidia-vaapi-driver/assets/110197801/e0d28f84-7977-4ce7-b2a9-b4c4c0409f70)
This section should be telling a lot if you send it here. Mine isn't what the result is supposed to be and I'm still trying to fix it, but just so you know what section to pay attention to.

**Comment by Vitaly76ASDgnaerkldbnak** [comment]
ff 110
![110](https://github.com/elFarto/nvidia-vaapi-driver/assets/120011445/aa519a70-86ba-49a4-a6cf-be01e3c0004c)
ff 125
![125](https://github.com/elFarto/nvidia-vaapi-driver/assets/120011445/5b7c7504-4d6c-4cb7-a014-169d3fdaf6de)

**Comment by Vitaly76ASDgnaerkldbnak** [comment]
Maybe this can help, I'm totally frustrated in this.

**Comment by mirh** [comment]
Obviously those two errors are either the usual [`widget.dmabuf.force-enabled`](https://github.com/elFarto/nvidia-vaapi-driver/issues/182) or [`media.hardware-video-decoding.force-enabled`](https://wiki.archlinux.org/title/firefox#Hardware_video_acceleration) thing.

**Comment by Vitaly76ASDgnaerkldbnak** [comment]
Even after enable [media.hardware-video-decoding.force-enabled](https://wiki.archlinux.org/title/firefox#Hardware_video_acceleration) in FF 125 there is no video acceleration. In FF 110 without this was. So it seems, it's something else...

**Comment by mirh** [comment]
Did you remember `MOZ_DISABLE_RDD_SANDBOX=1` and the dmabuf thing?

**Comment by Vitaly76ASDgnaerkldbnak** [comment]
Alas, but all this options are set in both versions of FF.

**Comment by Vitaly76ASDgnaerkldbnak** [comment]
In /etc/environment
``
NVD_BACKEND=direct
MOZ_DISABLE_RDD_SANDBOX=1
LIBVA_DRIVER_NAME=nvidia
``

**Comment by mirh** [comment]
Try firefox nightly
https://fedoraproject.org/wiki/How_to_debug_Firefox_problems#Testing_Mozilla_binaries

**Comment by Vitaly76ASDgnaerkldbnak** [comment]
I checked in FF 126.0b6 and it's works (stand along version). I think problem is in FF of kubuntu distribution, so it's no problem with nvidia-vaapi-driver. So the issue can be close, big thanks for helping me anyway.

**Comment by mirh** [comment]
Then it seems like you skipped *widget.dmabuf.force-enabled*...

**Comment by Vitaly76ASDgnaerkldbnak** [comment]
No, it is set true in FF 125. For me it's clear that FF in  ubuntu distributive in some way compiled without some essential thing that is not letting  HW acceleration to function, maybe it's connected to dmabuf, maybe to something else.  But when I download standalone version of FF 110 and later, than HW acceleration is here.

**Comment by mirh** [comment]
Wayland support has been built-in since at least some years [by now](https://bugzilla.mozilla.org/show_bug.cgi?id=1725245), if any it wasn't until [recent](https://www.phoronix.com/news/Ubuntu-23.10-Firefox-Wayland) that the snap enabled it by default (but that's kinda a different story). 

Speaking of ubuntu.. I just realized the version wasn't even reported here.

**Comment by Vitaly76ASDgnaerkldbnak** [comment]
I'm using FF from PPA https://ppa.launchpadcontent.net/mozillateam/ppa/ubuntu in Kubuntu 22.04.3 LTS, any versions of FF from 124.x to 125.x. And I'm in X11 not Wayland.

**Comment by mirh** [comment]
#210 had it working though?

**Comment by sekaiacg** [comment]
In addition to the recommended settings, I turned on hardware acceleration using the following additional settings on archlinux.
![image](https://github.com/elFarto/nvidia-vaapi-driver/assets/33929605/ad1b747d-b733-494e-9585-108c0160e212)

https://wiki.archlinux.org/title/firefox#Hardware_video_acceleration

**Comment by elFarto** [comment]
Assuming issue is resolved.
